packages/perseus/src/validation.types.ts

```diff
@@ -82,17 +81,26 @@ export type PerseusGradedGroupRubric = PerseusGradedGroupWidgetOptions;
 
 export type PerseusGradedGroupSetRubric = PerseusGradedGroupSetWidgetOptions;
 
-export type PerseusGrapherRubric = PerseusGrapherWidgetOptions;
+export type PerseusGrapherRubric = {
+    correct:
+        | {type: "absolute_value"; coords: [Coord, Coord]}
+        | {
+              type: "exponential";
+              asymptote: [Coord, Coord];
+              coords: [Coord, Coord];
+          }
+        | {type: "linear"; coords: [Coord, Coord]}
+        | {type: "logarithm"; asymptote: [Coord, Coord]; coords: [Coord, Coord]}
+        | {type: "quadratic"; coords: [Coord, Coord]}
+        | {type: "sinusoid"; coords: [Coord, Coord]}
+        | {type: "tangent"; coords: [Coord, Coord]};
+};
 
 /**
  * TODO: this is kind of just a guess right now
  * based off of an old comment in grapher
  */
-export type PerseusGrapherUserInput = {
-    type: string;
-    asymptote: ReadonlyArray<Coord>;
-    coords: ReadonlyArray<Coord>;
-};
+export type PerseusGrapherUserInput = PerseusGrapherRubric["correct"];
```

Comment:
I was thinking this might be enough to remove the TODO for this section. Wanted to get some input before removing it though. Thoughts?

Reply:
Yeah, I think you could remove the TODO